app/src/main/java/de/othr/plantico/ui/ownedPlantList/OwnedPlantListAdapter.kt

```diff
@@ -0,0 +1,163 @@
+@file:Suppress("SENSELESS_COMPARISON")
+
+package de.othr.plantico.ui.ownedPlantList
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.Intent
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import de.othr.plantico.R
+import de.othr.plantico.addDays
+import de.othr.plantico.database.entities.OwnedPlant
+import de.othr.plantico.database.entities.Plant
+import de.othr.plantico.database.entities.WateringLevel
+import de.othr.plantico.databinding.ViewOwnedPlantItemBinding
+import de.othr.plantico.nowUTC
+import de.othr.plantico.ui.PlantActivity
+import java.time.LocalDate
+import java.util.*
+
+
+class OwnedPlantListAdapter(context: Context) :
+    ListAdapter<OwnedPlant, OwnedPlantListAdapter.OwnedPlantViewHolder>(OwnedPlantsComparator()) {
+    private val con: Context = context
+    private var plantList: List<Plant>? = null
+
+    fun addPlants(plants: List<Plant>) {
```

Comment:
setPlants / setPlantList would be a more fitting name, since you aren't adding plants, but resetting the list.